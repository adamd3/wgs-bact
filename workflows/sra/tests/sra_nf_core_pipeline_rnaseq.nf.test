nextflow_workflow {

    name "Test workflow: sra/main.nf"
    script "../main.nf"
    workflow "SRA"
    tag "workflows"
    tag "workflows_sra"
    tag "multiqc_mappings_config"
    tag "sra_nf_core_pipeline_rnaseq"

    test("Parameters: --nf_core_pipeline rnaseq") {

        when {
            workflow {
                """
                input[0] = Channel.from("DRX026011", "ERX1234253", "SRX6725035")
                """
            }
            params {
                outdir = "$outputDir"
                nf_core_pipeline = "rnaseq"
            }
        }

        then {
            assert workflow.success

            assertAll(
                {
                    with(workflow.out.samplesheet) {
                        assert path(get(0)).readLines().size() == 4
                        assert path(get(0)).readLines()*.split(',')[0].take(5) == ['"sample"', '"fastq_1"', '"fastq_2"', '"strandedness"' , '"run_accession"']
                        assert path(get(0)).readLines()*.split(',').collect { it[0] } == ['"sample"', '"DRX026011"', '"ERX1234253"', '"SRX6725035"']
                        assert path(get(0)).text.contains('Illumina HiSeq 2500')
                    }
                },
                {
                    with(workflow.out.mappings) {
                        assert path(get(0)).readLines().size() == 4
                        assert path(get(0)).readLines()*.split(',').collect { it[0] } == ['"sample"', '"DRX026011"', '"ERX1234253"', '"SRX6725035"']
                        assert path(get(0)).text.contains('Illumina HiSeq 2500')
                    }
                },
                {
                    with(workflow.out.sample_mappings) {
                        assert path(get(0)[0]).md5 == "1ac06bb95b503703430e74660bbdd768"
                    }
                }
            )
        }
    }
}
