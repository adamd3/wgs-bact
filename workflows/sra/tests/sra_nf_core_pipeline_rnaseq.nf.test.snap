{
    "Parameters: --nf_core_pipeline rnaseq": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "samplesheet.csv:md5,69117245b73776d6e4daecdbfdff9767"
                ],
                "2": [
                    "id_mappings.csv:md5,a18766d39f6b2239163b4ef00eeaf342"
                ],
                "3": [
                    [
                        "multiqc_config.yml:md5,bb99e81ddd360988338c10ecbdbce65a",
                        "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9"
                    ]
                ],
                "4": [
                    "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9",
                    "versions.yml:md5,45e4a369df2d3af023b8622aa3f6d5a8",
                    "versions.yml:md5,9b17045ca8bdc272cb3f9d349a81d206",
                    "versions.yml:md5,b21338b74b9a8fe0c7a114f7686c07cd",
                    "versions.yml:md5,b52279f7d6b891a6523d9321f3f85b47",
                    "versions.yml:md5,e977678bd13a54fc9c31addf135be15e"
                ],
                "fastq": [
                    
                ],
                "mappings": [
                    "id_mappings.csv:md5,a18766d39f6b2239163b4ef00eeaf342"
                ],
                "sample_mappings": [
                    [
                        "multiqc_config.yml:md5,bb99e81ddd360988338c10ecbdbce65a",
                        "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9"
                    ]
                ],
                "samplesheet": [
                    "samplesheet.csv:md5,69117245b73776d6e4daecdbfdff9767"
                ],
                "versions": [
                    "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9",
                    "versions.yml:md5,45e4a369df2d3af023b8622aa3f6d5a8",
                    "versions.yml:md5,9b17045ca8bdc272cb3f9d349a81d206",
                    "versions.yml:md5,b21338b74b9a8fe0c7a114f7686c07cd",
                    "versions.yml:md5,b52279f7d6b891a6523d9321f3f85b47",
                    "versions.yml:md5,e977678bd13a54fc9c31addf135be15e"
                ]
            }
        ],
        "timestamp": "2023-10-06T12:14:02.500312"
    }
}